File: mario.c
#include <cs50.h>
#include <stdio.h>

void builder(int n);
int getInt(string prompt);
int checkInt(int n);

int main(void)
{
    //get int
    int blocks = getInt("pick an integer between 1 and 8 inclusive");
    
    //now print blocks
    builder(blocks);
}

void builder(int n)
{
    //building from top down I guess
    for(int i = 1; i < (n + 1); i++)
    {
        //adding spaces to the front
        if(n > 1)
        {
            int frontSpaces = n - i;
            for(int k = 0 ; k < frontSpaces; k++)
            {
                printf(" ");
            }
        }
        
        //printing 2*i+1 for each level (to include spaces)
        //on the last print ensure \n
        int printTimes = 2 * i + 1;
        for(int j = 1; j < printTimes + 1; j++)
        {
            if(j == ((printTimes + 1) / 2 ))
            {
                printf(" ");
            }
            if(j == printTimes)
            {
                printf("\n");
            }
            else
            {
                printf("#");
            }
        }
    }
}

int getInt(string prompt)
{
    
    int n;
    int check;
    do
    {
        n = get_int("%s\n", prompt);
        check = checkInt(n);
    }
    while (check == 0);
    return n;
}

int checkInt(int n)
{
    if(n < 1)
    {
        return 0;
    }
    if(n > 8)
    {
        return 0;
    }
    return 1;
}

File: mario.py
from cs50 import get_string
from cs50 import get_int


def main():

    # First prompt for size
    size = getSize()

    # Now build

    builder(size)

# prints out the blocks according to a given height


def builder(size):
    # Printing from the top down
    for i in range(1, size + 1):
        # Prints 2 * height + 2 times
        spaces = size - i

        for i in range(spaces):
            print(" ", end="")

        # Now print blocks
        blocks = size - spaces

        for i in range(blocks):
            print('#', end="")

        # Now print two middle spaces
        for i in range(2):
            print(' ', end="")

        # Now print blocks then spaces

        for i in range(blocks):
            print('#', end="")

        print()


def getSize():
    size = 0
    while size < 1 or size > 8:

        size = get_int("Give me an int between 1 and 8 inclusive  ")

    return size


main()